FirebaseAuth/Sources/Auth/FIRAuth.m

```diff
@@ -1867,7 +2007,51 @@ - (void)internalCreateUserWithEmail:(NSString *)email
     completion(nil, [FIRAuthErrorUtils missingEmailErrorWithMessage:nil]);
     return;
   }
+
+#if TARGET_OS_IOS && !TARGET_OS_MACCATALYST
+  if ([[FIRAuthRecaptchaVerifier sharedRecaptchaVerifier:self]
+          enablementStatusForProvider:FIRAuthRecaptchaProviderPassword]) {
+    [[FIRAuthRecaptchaVerifier sharedRecaptchaVerifier:self]
+        injectRecaptchaFields:request
+                     provider:FIRAuthRecaptchaProviderPassword
+                       action:FIRAuthRecaptchaActionSignUpPassword
+                   completion:^(
+                       FIRIdentityToolkitRequest<FIRAuthRPCRequest> *requestWithRecaptchaToken) {
+                     [FIRAuthBackend
+                         signUpNewUser:(FIRSignUpNewUserRequest *)requestWithRecaptchaToken
+                              callback:completion];
+                   }];
+  } else {
```

Comment:
Will the recaptchaVersion and recaptchaClient fields be included in the request even if recaptcha enterprise is not enabled? We need this for the autodetect capability, right?

Reply:
> Will the recaptchaVersion and recaptchaClient fields be included in the request even if recaptcha enterprise is not enabled?

Nope. See line 2053.

For autodetect, handled at line  2036.
For pre-initialization, line 2012.

Reply:
We enter line 2036 only if we send an initial request and get back MISSING_RECAPTCHA_TOKEN?

Reply:
Yeah, correct.

Reply:
Synced offline. clientType is always set here: https://github.com/firebase/firebase-ios-sdk/blob/835dbbcff034fa1dbf5fcdd170b02fa42c108998/FirebaseAuth/Sources/Backend/FIRIdentityToolkitRequest.m#L55